File: server/module/module-mall/mall-server/src/convert/mall_promotion_seckill_product.rs
use sea_orm::{Set, NotSet};
use crate::model::mall_promotion_seckill_product::{self, Model as MallPromotionSeckillProduct, ActiveModel as MallPromotionSeckillProductActiveModel};
use mall_model::request::mall_promotion_seckill_product::{CreateMallPromotionSeckillProductRequest, UpdateMallPromotionSeckillProductRequest};
use mall_model::response::mall_promotion_seckill_product::MallPromotionSeckillProductResponse;

pub fn create_request_to_model(request: &CreateMallPromotionSeckillProductRequest) -> MallPromotionSeckillProductActiveModel {
    MallPromotionSeckillProductActiveModel {
        activity_id: Set(request.activity_id.clone()),
        config_ids: Set(request.config_ids.clone()),
        spu_id: Set(request.spu_id.clone()),
        sku_id: Set(request.sku_id.clone()),
        seckill_price: Set(request.seckill_price.clone()),
        stock: Set(request.stock.clone()),
        activity_status: Set(request.activity_status.clone()),
        activity_start_time: Set(request.activity_start_time.clone()),
        activity_end_time: Set(request.activity_end_time.clone()),
        ..Default::default()
    }
}

pub fn update_request_to_model(request: &UpdateMallPromotionSeckillProductRequest, existing: MallPromotionSeckillProduct) -> MallPromotionSeckillProductActiveModel {
    let mut active_model: MallPromotionSeckillProductActiveModel = existing.into();
    if let Some(activity_id) = &request.activity_id { 
        active_model.activity_id = Set(activity_id.clone());
    }
    if let Some(config_ids) = &request.config_ids { 
        active_model.config_ids = Set(config_ids.clone());
    }
    if let Some(spu_id) = &request.spu_id { 
        active_model.spu_id = Set(spu_id.clone());
    }
    if let Some(sku_id) = &request.sku_id { 
        active_model.sku_id = Set(sku_id.clone());
    }
    if let Some(seckill_price) = &request.seckill_price { 
        active_model.seckill_price = Set(seckill_price.clone());
    }
    if let Some(stock) = &request.stock { 
        active_model.stock = Set(stock.clone());
    }
    if let Some(activity_status) = &request.activity_status { 
        active_model.activity_status = Set(activity_status.clone());
    }
    if let Some(activity_start_time) = &request.activity_start_time { 
        active_model.activity_start_time = Set(activity_start_time.clone());
    }
    if let Some(activity_end_time) = &request.activity_end_time { 
        active_model.activity_end_time = Set(activity_end_time.clone());
    }
    active_model
}

pub fn model_to_response(model: MallPromotionSeckillProduct) -> MallPromotionSeckillProductResponse {
    MallPromotionSeckillProductResponse { 
        id: model.id,
        activity_id: model.activity_id,
        config_ids: model.config_ids,
        spu_id: model.spu_id,
        sku_id: model.sku_id,
        seckill_price: model.seckill_price,
        stock: model.stock,
        activity_status: model.activity_status,
        activity_start_time: model.activity_start_time,
        activity_end_time: model.activity_end_time,
        creator: model.creator,
        create_time: model.create_time,
        updater: model.updater,
        update_time: model.update_time,
    }
}

File: admin/src/api/mall_promotion_seckill_product.ts
import { PaginatedRequest, PaginatedResponse } from '@/base/page';
import { api } from '@/utils/request';

const apis = {
  create: '/erp/mall_promotion_seckill_product/create', // 新增
  update: '/erp/mall_promotion_seckill_product/update', // 修改
  delete: '/erp/mall_promotion_seckill_product/delete', // 删除
  get: '/erp/mall_promotion_seckill_product/get', // 单条查询
  list: '/erp/mall_promotion_seckill_product/list', // 列表查询
  page: '/erp/mall_promotion_seckill_product/page', // 分页查询
}

export interface MallPromotionSeckillProductRequest {
  id: number; // 秒杀参与商品编号
  activity_id: number; // 秒杀活动 id
  config_ids: string; // 秒杀时段 id 数组
  spu_id: number; // 商品 spu_id
  sku_id: number; // 商品 sku_id
  seckill_price: number; // 秒杀金额，单位：分
  stock: number; // 秒杀库存
  activity_status: number; // 秒杀商品状态
  activity_start_time: string; // 活动开始时间点
  activity_end_time: string; // 活动结束时间点
  }

export interface MallPromotionSeckillProductResponse {
  id: number; // 秒杀参与商品编号
  activity_id: number; // 秒杀活动 id
  config_ids: string; // 秒杀时段 id 数组
  spu_id: number; // 商品 spu_id
  sku_id: number; // 商品 sku_id
  seckill_price: number; // 秒杀金额，单位：分
  stock: number; // 秒杀库存
  activity_status: number; // 秒杀商品状态
  activity_start_time: string; // 活动开始时间点
  activity_end_time: string; // 活动结束时间点
  creator: number; // 创建者ID
  create_time: string; // 创建时间
  updater: number; // 更新者ID
  update_time: string; // 更新时间
  }

export interface MallPromotionSeckillProductQueryCondition extends PaginatedRequest {
  
}

export const createMallPromotionSeckillProduct = (mall_promotion_seckill_product: MallPromotionSeckillProductRequest): Promise<number> => {
  return api.post<number>(apis.create, mall_promotion_seckill_product);
}

export const updateMallPromotionSeckillProduct = (mall_promotion_seckill_product: MallPromotionSeckillProductRequest): Promise<void> => {
  return api.post<void>(apis.update, mall_promotion_seckill_product);
}

export const deleteMallPromotionSeckillProduct = (id: number): Promise<void> => {
  return api.post<void>(`${apis.delete}/${id}`);
}

export const getMallPromotionSeckillProduct = (id: number): Promise<MallPromotionSeckillProductResponse> => {
  return api.get<MallPromotionSeckillProductResponse>(`${apis.get}/${id}`);
}

export const listMallPromotionSeckillProduct = (): Promise<Array<MallPromotionSeckillProductResponse>> => {
  return api.get<Array<MallPromotionSeckillProductResponse>>(apis.list);
}

export const pageMallPromotionSeckillProduct = (condition: MallPromotionSeckillProductQueryCondition): Promise<PaginatedResponse<MallPromotionSeckillProductResponse>> => {
  return api.get<PaginatedResponse<MallPromotionSeckillProductResponse>>(apis.page, condition);
}
